### PR TITLE
Make VideoServer and VideoPlayer non-singletons

### PR DESCRIPTION
**Pull request by LipuFei**

### CODE REVIEW

**Comment by LipuFei** [comment]
retest this please

**Comment by LipuFei** [comment]
retest this please

**Comment by LipuFei** [comment]
@whirm ready

**Comment by whirm** [comment]
@LipuFei needs rebase

**Comment by LipuFei** [comment]
retest this please

**Comment by LipuFei** [comment]
retest this please

**Comment by LipuFei** [comment]
retest this please

**Comment by LipuFei** [comment]
retest this please

**Comment by NielsZeilemaker** [comment]
retest this please

**Comment by LipuFei** [comment]
@whirm ready

File: Tribler/Main/vwxGUI/SearchGridManager.py [hunk 1]
```diff
@@ -635,7 +634,7 @@ def _get_videoplayer(self):
         """
         Returns the VideoPlayer instance.
         """
-        return VideoPlayer.getInstance()
+        return self.guiUtility.utility.session.lm.videoplayer
```
Comment (NielsZeilemaker) [inline Tribler/Main/vwxGUI/SearchGridManager.py]: I would use self.guiUtility.videoplayer

File: Tribler/Main/vwxGUI/list_details.py [hunk 2]
```diff
@@ -2378,7 +2377,7 @@ def RemoveFileindex(self, fileindex):
                 self.vSizer.Remove(link)
                 self.OnChange()
 
-        vod_dl = VideoPlayer.getInstance().get_vod_download()
+        vod_dl = self.guiutility.utility.session.lm.videoplayer.get_vod_download()
```
Comment (NielsZeilemaker) [inline Tribler/Main/vwxGUI/list_details.py]: Same here

**Comment by NielsZeilemaker** [comment]
The rest seems fine, you might want to consider implementing a get_video_player method in session, but thats up to you.

**Comment by LipuFei** [comment]
@NielsZeilemaker Will do. This is still an ugly fix. We plan to make a GUI-less Session Core which has all the functions we need. This is just step one.

**Comment by NielsZeilemaker** [comment]
Still a step in the right direction

**Comment by LipuFei** [comment]
@whirm ready

**Pull request merged**